libimage/pull.go

```diff
@@ -516,6 +600,9 @@ func (r *Runtime) copySingleImageFromRegistry(ctx context.Context, imageName str
 	}
 
 	sys := r.systemContextCopy()
+	if options.shortNameMode != nil {
+		sys.ShortNameMode = options.shortNameMode
```

Comment:
(Absolutely non-blocking: Alternatively, this could be
```go
if options.copyFromRegistryDryRun {
    sys.ShortNameMode = &shortNameModeDisabled
}
```

+ rename `copyFromRegistryDryRun` to something like `onlyResolveRegistryImageName`; that would mean one new field in `options` instead of two, and more closely express what we are doing with that option — _assuming_ nothing else would ever need to actually use `options.shortNameMode` for other purposes.